department-of-veterans-affairs/caseflow
# Use FeatureToggle for non-prod Uploads

Resolves [Create Feature Toggle for Claim Evidence Response](https://jira.devops.va.gov/browse/APPEALS-37637)

# Description
- When not in production, use a new feature toggle to control the success/failure of the ClaimEvidenceService document upload for correspondences vs.  the old, hard-coded boolean that was being used.

- Add spec to test upload behavior.

- Fix a few misc. issues around some missed requirements for the parent ticket's work.

## Acceptance Criteria

## Testing Plan
1. Run new spec and verify it passes.
2. Toggle flag _off_ and verify:
    1. Intake form submit fails
    2. A `EfolderUploadFailedTask` is created with a status of `in progress`.
3. Toggle flag _on_ and verify:
    1. Intake form successfully submits and creates selected tasks.
